oak_functions/examples/key_value_lookup/module/src/tests.rs

```diff
@@ -15,85 +15,40 @@
 
 #[tokio::test]
 async fn test_server() {
-    let server_port = test_utils::free_port();
-    let address = SocketAddr::from((Ipv6Addr::UNSPECIFIED, server_port));
-
-    let mut manifest_path = std::env::current_dir().unwrap();
-    manifest_path.push("Cargo.toml");
-
-    let wasm_module_bytes =
-        test_utils::compile_rust_wasm(manifest_path.to_str().expect("Invalid target dir"), false)
-            .expect("Couldn't read Wasm module");
+    let wasm_path = test_utils::build_rust_crate_wasm("key_value_lookup").unwrap();
 
-    let mock_static_server = Arc::new(test_utils::MockStaticServer::default());
+    let lookup_data_file =
+        test_utils::write_to_temp_file(&test_utils::serialize_entries(hashmap! {
+            b"key_0".to_vec() => b"value_0".to_vec(),
+            b"key_1".to_vec() => b"value_1".to_vec(),
+            b"key_2".to_vec() => b"value_2".to_vec(),
+            b"empty".to_vec() => vec![],
+        }));
 
-    let mock_static_server_clone = mock_static_server.clone();
-    let static_server_port = test_utils::free_port();
-    let mock_static_server_background = test_utils::background(|term| async move {
-        mock_static_server_clone
-            .serve(static_server_port, term)
-            .await
-    });
+    let server_port = test_utils::free_port();
+    let server_background = test_utils::create_and_start_oak_functions_server(
+        server_port,
+        &wasm_path,
+        lookup_data_file.path().to_str().unwrap(),
+    )
+    .unwrap();
 
-    mock_static_server.set_response_body(test_utils::serialize_entries(hashmap! {
-        b"key_0".to_vec() => b"value_0".to_vec(),
-        b"key_1".to_vec() => b"value_1".to_vec(),
-        b"key_2".to_vec() => b"value_2".to_vec(),
-        b"empty".to_vec() => vec![],
-    }));
-
-    let policy = ServerPolicy {
-        constant_response_size_bytes: 100,
-        constant_processing_time_ms: 200,
-    };
-    let logger = Logger::for_test();
-    let lookup_data_manager = Arc::new(LookupDataManager::new_empty(logger.clone()));
-    let lookup_data_refresher = LookupDataRefresher::new(
-        Some(LookupDataSource::Http {
-            url: format!("http://localhost:{}", static_server_port),
-            auth: LookupDataAuth::default(),
-        }),
-        lookup_data_manager.clone(),
-        logger.clone(),
-    );
-    lookup_data_refresher.refresh().await.unwrap();
-
-    let lookup_factory = LookupFactory::new_boxed_extension_factory(lookup_data_manager)
-        .expect("could not create LookupFactory");
-
-    let wasm_handler =
-        create_wasm_handler(&wasm_module_bytes, vec![lookup_factory], logger.clone())
-            .expect("could not create wasm_handler");
-
-    let server_background = test_utils::background(|term| async move {
-        create_and_start_grpc_server(&address, wasm_handler, policy.clone(), term, logger).await
-    });
+    // Wait for the server to start up.
+    std::thread::sleep(Duration::from_secs(2));
```

Comment:
Is there any way we could wait for `create_and_start_oak_functions_server` instead of `sleep`ing?

Reply:
I don't think so, this is an opaque child process listening on the given port. I guess we could try and probe that port in a loop until it's open, then close that connection and allow the rest of the code to connect to it at that point, but it would still involve some polling / sleeping anyways.